### PR TITLE
Add Integration tests to CI

### PR DESCRIPTION
**Pull request by adamhathcock**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    image: speckle/speckle-preview-service:latest
+    restart: always
+    depends_on:
+      speckle-server:
+        condition: service_healthy
+    environment:
+      DEBUG: "preview-service:*"
+      PG_CONNECTION_STRING: "postgres://speckle:speckle@postgres/speckle"
+
+  webhook-service:
```
[the remark this answers is not in the document]
Comment (JR-Morgan) [reply docker-compose.yml:125]: We are not

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    restart: always
+    depends_on:
+      speckle-server:
+        condition: service_healthy
+    environment:
+      DEBUG: "webhook-service:*"
+      PG_CONNECTION_STRING: "postgres://speckle:speckle@postgres/speckle"
+      WAIT_HOSTS: postgres:5432
+
+  fileimport-service:
```
[the remark this answers is not in the document]
Comment (JR-Morgan) [reply docker-compose.yml:136]: We are not

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      DEBUG: "speckle:*"
+
+      POSTGRES_URL: "postgres"
+      POSTGRES_USER: "speckle"
+      POSTGRES_PASSWORD: "speckle"
+      POSTGRES_DB: "speckle"
+      ENABLE_MP: "false"
+
+  preview-service:
+    image: speckle/speckle-preview-service:latest
```
[the remark this answers is not in the document]
Comment (JR-Morgan) [reply docker-compose.yml:116]: I have a suspicion we don't need the preview service.
A few graphql endpoints return the preview URL, and some assertions on that property.
Unsure if that url comes from the preview service, or if it is set before...

Either way, I think we should remove, and and we can always remove those assertions if we find these tests failing.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        ]
+      interval: 5s
+      timeout: 30s
+      retries: 30
+      start_period: 10s
+
+  ####
+  # Speckle Server
+  #######
+  speckle-frontend:
```
[the remark this answers is not in the document]
Comment (JR-Morgan) [reply docker-compose.yml:55]: I assume not, if its just the frontend, and has no rest endpoints that we are depending on... I think we can remove it.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    healthcheck:
+      test:
+        - CMD
+        - node
+        - -e
+        - "try { require('node:http').request({headers: {'Content-Type': 'application/json'}, port:3000, hostname:'127.0.0.1', path:'/liveness', method: 'GET', timeout: 2000 }, (res) => { body = ''; res.on('data', (chunk) => {body += chunk;}); res.on('end', () => {process.exit(res.statusCode != 200 || body.toLowerCase().includes('error'));}); }).end(); } catch { process.exit(1); }"
+      interval: 10s
+      timeout: 10s
+      retries: 3
+      start_period: 90s
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply docker-compose.yml:68]: I just copy/pasted from the docker-compose you linked really.   but that makes sense

[the remark this answers is not in the document]
Comment (adamhathcock) [reply docker-compose.yml:68]: doesn't a longer start period just delay the health check but not start up?  

Yeah it's fine:
https://docs.docker.com/reference/dockerfile/#healthcheck
start period provides initialization time for containers that need time to bootstrap. Probe failure during that period will not be counted towards the maximum number of retries. However, if a health check succeeds during the start period, the container is considered started and all consecutive failures will be counted towards the maximum number of retries.

**Pull request merged**